tests/conftest.py

```diff
@@ -204,3 +205,112 @@ def _R(expr):
         assert len(base) == 1
         base = base.pop()
     return EvalDerivative(*expr.args, base=base)
+
+
+# Utilities for testing tree structure
+
+
+def assert_structure(operator, exp_trees=None, exp_iters=None):
+    """
+    Utility function that helps to check loop structure of IETs. Retrieves trees from an
+    Operator and check that the blocking structure is as expected. Trees and Iterations
+    are returned for further use in tests.
+
+    Examples
+    --------
+    To check that an Iteration tree has the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x
+                for y
+            for f
+                for y
+
+    we call:
+
+    .. code-block:: python
+
+        trees, iters = assert_structure(op, ['t,x,y', 't,f,y'], 't,x,y,f,y')`
```

Comment:
`t` or `time` ?

Reply:
t to save space as the mapper below at line 237 takes care

Reply:
I get that but it's not consitent with the code block above that doesn't have `t` so I would use `time`

Reply:
Right, this is an example and the user should use 't' instead of time. So I cannot change the example. I added a note. Maybe it is better? Lemme think how else to improve it.

Reply:
just change the example from `for time` to `for t` ?
anyway, do I get it right: it would work with both `t` and `time` in practice, right?